src/SignalR/clients/ts/signalr/tests/LongPollingTransport.test.ts

```diff
@@ -165,6 +165,59 @@ describe("LongPollingTransport", () => {
             expect(secondPollUserAgent).toEqual(value);
         });
     });
+
+    it("overwrites library headers with user headers", async () => {
+        await VerifyLogger.run(async (logger) => {
+            const headers = { "User-Agent": "Custom Agent", "X-HEADER": "VALUE" };
+            let firstPoll = true;
+            let firstPollUserAgent = "";
+            let firstUserHeader = "";
+            let secondPollUserAgent = "";
+            let secondUserHeader = "";
+            let deleteUserAgent = "";
+            let deleteUserHeader = "";
+            const pollingPromiseSource = new PromiseSource();
+            const httpClient = new TestHttpClient()
+                .on("GET", async (r) => {
+                    if (firstPoll) {
+                        firstPoll = false;
+                        firstPollUserAgent = r.headers!["User-Agent"];
+                        firstUserHeader = r.headers!["X-HEADER"];
+                        return new HttpResponse(200);
+                    } else {
+                        secondPollUserAgent = r.headers!["User-Agent"];
+                        secondUserHeader = r.headers!["X-HEADER"];
+                        await pollingPromiseSource.promise;
+                        return new HttpResponse(204);
+                    }
+                })
+                .on("DELETE", async (r) => {
+                    deleteUserAgent = r.headers!["User-Agent"];
+                    deleteUserHeader = r.headers!["X-HEADER"];
+                    return new HttpResponse(202);
+                });
+
+            const transport = new LongPollingTransport(httpClient, undefined, logger, false, true, headers);
+
+            await transport.connect("http://tempuri.org", TransferFormat.Text);
+
```

Comment:
can you add a `transport.send` and check the headers for the "POST" on `TestHttpClient`

Reply:
Fixed: e47849d1fbfb8c12711e634c69c74fcca5e6c919